test/language/expressions/assignment/destructuring/array-elem-put-constant-no-strict.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    The assignment target should obey `const` semantics outside of strict mode.
+es6id: 12.14.5.3
+features: [const]
+flags: [noStrict]
+---*/
+
+var value = [1];
+const c = null;
+var result;
+
+result = [ c ] = value;
```

Comment:
> If I understand correctly, then this test could be made valid if the constants were not declared in the global environment. Is that right?

`const` declarations are always created as immutable strict bindings, so even if the test is wrapped in a function or executed in eval- or module-code, an assignment to the binding throws a `TypeError`. Or do you mean something else?

Spec references:
9.2.12 FunctionDeclarationInstantiation(func, argumentsList), step 35.b.i.1
15.1.8 Runtime Semantics: GlobalDeclarationInstantiation (script, env), step 15.b.i.1
15.2.1.16.4 ModuleDeclarationInstantiation( ) Concrete Method, step 16.a.i.1
18.2.1.2 Runtime Semantics: EvalDeclarationInstantiation( body, varEnv, lexEnv, strict), step 14.b.i.1


> I think a better fix would be to collapse all const-strict/const-no-strict file pairs and omit the onlyStrict/noStrict flag entirely.

Yes, that seems reasonable.